### ISSUE TITLE
pg_format exited with status code 65280

### ISSUE THREAD

**Issue by mxgrn** [akoutmos/ecto_dbg#7]
On a somewhat complex query, I get the following crash:

```log
[error] ** (FunctionClauseError) no function clause matching in String.split/3
    (elixir 1.17.2) lib/string.ex:489: String.split({:error, "pg_format exited with status code 65280\nUncaught exception: Use of uninitialized value in pattern match (m//) at /Users/mxgrn/code/mxgrn/ketoketo/deps/ecto_dbg/priv/lib/pgFormatter/Beautify.pm line 965, <STDIN> line 1.\n at /Users/mxgrn/code/mxgrn/ketoketo/_build/dev/lib/ecto_dbg/priv/pg_format line 19, <STDIN> line 1.\n\tmain::__ANON__(\"Use of uninitialized value in pattern match (m//) at /Users/m\"...) called at /Users/mxgrn/code/mxgrn/ketoketo/deps/ecto_dbg/priv/lib/pgFormatter/Beautify.pm line 965\n\tpgFormatter::Beautify::beautify(pgFormatter::Beautify=HASH(0x12e0117c0)) called at /Users/mxgrn/code/mxgrn/ketoketo/deps/ecto_dbg/priv/lib/pgFormatter/CLI.pm line 170\n\tpgFormatter::CLI::beautify(pgFormatter::CLI=HASH(0x12d80c620)) called at /Users/mxgrn/code/mxgrn/ketoketo/deps/ecto_dbg/priv/lib/pgFormatter/CLI.pm line 78\n\tpgFormatter::CLI::run(pgFormatter::CLI=HASH(0x12d80c620)) called at /Users/mxgrn/code/mxgrn/ketoketo/_build/dev/lib/ecto_dbg/priv/pg_format line 49"}, "\n", [])
    (ecto_dbg 0.2.0) lib/ecto_dbg.ex:161: EctoDbg.format_log_message/1
    (ecto_dbg 0.2.0) lib/ecto_dbg.ex:141: EctoDbg.log_query/4
    (ecto_dbg 0.2.0) lib/ecto_dbg.ex:153: EctoDbg.run_and_log_query/5
```

The query, before calling `Repo.all_and_log`, reads as following via dbg:

```elixir
#Ecto.Query<from p0 in Ketoketo.Schemas.Product.Menu.Uk.Product,
 left_join: p1 in subquery(from p0 in Ketoketo.Site.ProductImage,
  order_by: [asc: p0.product_id, desc: p0.inserted_at],
  distinct: [asc: p0.product_id]),
 on: p0.id == p1.product_id, left_join: i2 in Ketoketo.Site.Image,
 on: i2.id == p1.image_id, join: m3 in assoc(p0, :menu_category), as: :mc,
 join: a4 in assoc(p0, :article_products), join: a5 in assoc(a4, :article),
 on: is_nil(a5.archived_at) and a5.published,
 join: a6 in assoc(a5, :article_products), join: p7 in assoc(a6, :product),
 left_join: p8 in assoc(p7, :product_stocks),
 left_join: l9 in assoc(p7, :location_products),
 left_join: o10 in Ketoketo.Schemas.OrderItem.Default.OrderItem,
 on: o10.article_id == a5.id and o10.order_id == ^0,
 join: p11 in assoc(p0, :product_ingredients),
 join: i12 in assoc(p11, :ingredient),
 where: p0.type == "food" and p0.published and is_nil(p0.archived_at),
 where: is_nil(m3.audience_id),
 select: %Ketoketo.Schemas.Product.Menu.Uk.Product{p0 | head_image: i2},
 preload: [
  articles:
    {a5,
     article_products: {a6, product: {p7, location_products: l9, product_stocks: p8}},
     order_items: o10},
  product_ingredients: {p11, ingredient: i12}
]>
```

**Comment by akoutmos** [comment]
Unfortunately for large queries, the SQL formatter that I am currently using dies...and it looks like your query may end up being pretty large with all of the joins. I am working on switching over to a Rust NIF for the formatting so hopefully that fixes this issue. Should have that done in a week or so!

**Comment by akoutmos** [comment]
This should be fixed in version `0.3.0`!
